### PR TITLE
Change translation of __NULL literal in bytecode generation phase

### PR DESCRIPTION
**Pull request by jneira**
To fix the bug #268 
I've tested manually with this [code](https://github.com/jneira/eta-test/blob/master/app/Main.hs):
```haskell
main = do
  putStrLn $ "nullPtr: " ++ show nullPtr
  let (fptr,offset,length) = BSInt.toForeignPtr B.empty
  withForeignPtr fptr $ \ ptr -> do
    trace ptr offset length
    putStrLn $ "ptr: " ++ show ptr
    putStrLn $ "ptr == nullPtr " ++ (show $ nullPtr == ptr)
  putStrLn $ "ByteString.empty: " ++  show B.empty
  putStrLn $ "Wai request: " ++ show defaultRequest
```
with this output:
```
Ptr: java.nio.HeapByteBuffer[pos=4 lim=4 cap=4]
Ofset: 0
Length: 0
nullPtr: 0xffffffff
ptr: 0xffffffff
ptr == nullPtr True
ByteString.empty: ""
Wai request: Request {requestMethod = "GET", httpVersion = HTTP/1.0, rawPathInfo = "", rawQueryString = "", requestHeaders = [], isSecure = False, remoteHost = 0.0.0.0:0, pathInfo = [], queryString = [], requestBody = <IO ByteString>, vault = <Vault>, requestBodyLength = KnownLength 0, requestHeaderHost = Nothing, requestHeaderRange = Nothing}
```

### CODE REVIEW

File: compiler/ETA/CodeGen/Utils.hs [hunk 1]
```diff
@@ -24,7 +25,9 @@ cgLit (MachWord64 i)        = (jlong, lconst $ fromIntegral i)
 cgLit (MachFloat r)         = (jfloat, fconst $ fromRational r)
 cgLit (MachDouble r)        = (jdouble, dconst $ fromRational r)
 -- TODO: Remove this literal variant?
-cgLit MachNullAddr          = (jobject, aconst_null jobject)
+cgLit MachNullAddr          = (jobject, nullAddr)
+  where nullAddr = getstatic $ mkFieldRef "eta/runtime/io/MemoryManager"
+                   "nullAddress" (obj "java/nio/ByteBuffer")
```
Comment (rahulmutt) [inline compiler/ETA/CodeGen/Utils.hs]: Couple changes:
- `(obj "java/nio/ByteBuffer")` -> `byteBufferType` (defined in `ETA.CodeGen.Rts`)
- `"eta/runtime/io/MemoryManager"` -> `memoryManager` (defined in `ETA.CodeGen.Rts`)

**Comment by rahulmutt** [comment]
This change will break the handling of `Maybe` arguments in the FFI (probably not used by anyone so probably no one will see the breakage?), because it does a (horrible) unsafeCoerce of `nullAddr#` in order to get `null`. I'll go ahead and file an issue tracking that breakage and we'll eventually get around to fixing it. I'll merge this pull request once it passed the build for the time being.

File: compiler/ETA/CodeGen/Utils.hs (hunk 1 above)
Comment (jneira) [inline compiler/ETA/CodeGen/Utils.hs]: mmm ok, otho you think would be better to use a static method to get the constant?

**Comment by jneira** [comment]
finally i've thought that the better place to set the `nullAddr :: Code` is in Rts itself

**Pull request merged**